test/integration/webhook_test.go

```diff
@@ -317,27 +317,31 @@ func TestWebhookGitHubPushWithImageStream(t *testing.T) {
 	}
 	defer watch.Stop()
 
-	for _, s := range []string{
-		"/oapi/v1/namespaces/" + testutil.Namespace() + "/buildconfigs/pushbuild/webhooks/secret101/github",
-	} {
+	s := "/oapi/v1/namespaces/" + testutil.Namespace() + "/buildconfigs/pushbuild/webhooks/secret101/github"
 
-		// trigger build event sending push notification
-		postFile(clusterAdminBuildClient.Build().RESTClient(), githubHeaderFunc, "github/testdata/pushevent.json", clusterAdminClientConfig.Host+s, http.StatusOK, t)
+	// trigger build event sending push notification
+	postFile(clusterAdminBuildClient.Build().RESTClient(), githubHeaderFunc, "github/testdata/pushevent.json", clusterAdminClientConfig.Host+s, http.StatusOK, t)
 
-		event := <-watch.ResultChan()
-		actual := event.Object.(*buildapi.Build)
-
-		// FIXME: I think the build creation is fast and in some situlation we miss
-		// the BuildPhaseNew here. Note that this is not a bug, in future we should
-		// move this to use go routine to capture all events.
-		if actual.Status.Phase != buildapi.BuildPhaseNew && actual.Status.Phase != buildapi.BuildPhasePending {
-			t.Errorf("Expected %s or %s, got %s", buildapi.BuildPhaseNew, buildapi.BuildPhasePending, actual.Status.Phase)
-		}
+	var build *buildapi.Build
 
-		if actual.Spec.Strategy.SourceStrategy.From.Name != "registry:3000/integration/imagestream:success" {
-			t.Errorf("Expected %s, got %s", "registry:3000/integration/imagestream:success", actual.Spec.Strategy.SourceStrategy.From.Name)
+Loop:
```

Comment:
looks ugly :)

you could use:
```
function() {
for {
  ...
  return
}
}()
```

Reply:
to me *that* looks ugly (nested anonymous function to handle control flow?).  beauty is in the eye of the beholder I suppose, but I feel like i'm actually using control structures provided by the language as they were intended to be used.